domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/metadata/ClaimantResponseMetadata.java

```diff
@@ -0,0 +1,105 @@
+    private final DecisionType courtDecision;
+    private final FormaliseOption formaliseOption;
+    private final Boolean mediation;
+
+    static ClaimantResponseMetadata fromClaim(Claim claim) {
+        final Optional<ClaimantResponse> optionalClaimantResponse = claim.getClaimantResponse();
+        if (!optionalClaimantResponse.isPresent()) {
+            return null;
+        }
+
+        final ClaimantResponse claimantResponse = optionalClaimantResponse.get();
+        final LocalDateTime claimantRespondedAt = claim.getClaimantRespondedAt().orElse(null);
+        final Boolean settleForAmount = extractBooleanOrNull(claimantResponse::getSettleForAmount, YesNoOption.YES);
+        final Boolean fullPaymentReceived = extractBooleanOrNull(
+            claimantResponse::getAmountPaid,
+            claim.getTotalAmountTillToday().orElse(null)
+        );
+        final ClaimantResponseType claimantResponseType = claimantResponse.getType();
+
+        switch (claimantResponseType) {
+            case ACCEPTATION:
+                final ResponseAcceptation responseAcceptation = (ResponseAcceptation) claimantResponse;
+                final PaymentPlanMetadata claimantPaymentPlan = responseAcceptation.getClaimantPaymentIntention()
+                    .map(PaymentPlanMetadata::fromPaymentIntention)
+                    .orElse(null);
+                final PaymentPlanMetadata courtPaymentPlan = responseAcceptation.getCourtDetermination()
+                    .map(CourtDetermination::getCourtPaymentIntention)
+                    .map(PaymentPlanMetadata::fromPaymentIntention)
+                    .orElse(null);
+                final DecisionType courtDecision = responseAcceptation.getCourtDetermination()
+                    .map(CourtDetermination::getDecisionType)
+                    .orElse(null);
+                final FormaliseOption formaliseOption = responseAcceptation.getFormaliseOption().orElse(null);
+
+                return new ClaimantResponseMetadata(
+                    claimantRespondedAt,
+                    claimantResponseType,
+                    fullPaymentReceived,
+                    settleForAmount,
+                    claimantPaymentPlan,
+                    courtPaymentPlan,
+                    courtDecision,
+                    formaliseOption,
+                    null
+                );
+
+            case REJECTION:
+                final ResponseRejection responseRejection = (ResponseRejection) claimantResponse;
+                final Boolean mediation = extractBooleanOrNull(responseRejection::getFreeMediation, YesNoOption.YES);
+                return new ClaimantResponseMetadata(
+                    claimantRespondedAt,
+                    claimantResponseType,
+                    fullPaymentReceived,
+                    settleForAmount,
+                    null,
+                    null,
+                    null,
+                    null,
```

Comment:
Rather than passing nulls in constructor, would be nice to use builder in these scenarios